experiments/download_ssl4eo.py

```diff
@@ -277,69 +205,36 @@ def get_patch_s2(
 
 def get_random_patches_match(
     idx: int,
-    collections: Dict[str, Any],
-    bands: Dict[str, Any],
+    collection: ee.ImageCollection,
+    bands: List[str],
     crops: Dict[str, Any],
+    dtype: str,
     dates: List[Any],
-    radius: int,
+    radius: float,
     debug: bool = False,
     match_coords: Dict[str, Any] = {},
-) -> Tuple[
-    Optional[List[Dict[str, Any]]],
-    Optional[List[Dict[str, Any]]],
-    Optional[List[Dict[str, Any]]],
-    List[float],
-]:
+) -> Tuple[Optional[List[Dict[str, Any]]], List[float]]:
     # (lon,lat) of idx patch
     coords = match_coords[str(idx)]
 
-    # random +- 15 days of random days within 1 year from the reference dates
-    delta = timedelta(days=np.random.randint(365))
-    periods = [get_period(date - delta, days=30) for date in dates]
-
-    collection_s1 = collections["s1_grd"]
-    collection_s2c = collections["s2_l1c"]
-    collection_s2a = collections["s2_l2a"]
-
-    bands_s1 = bands["s1_grd"]
-    bands_s2c = bands["s2_l1c"]
-    bands_s2a = bands["s2_l2a"]
-
-    crop_s1 = crops["s1_grd"]
-    crop_s2c = crops["s2_l1c"]
-    crop_s2a = crops["s2_l2a"]
+    # random +- 30 days of random days within 1 year from the reference dates
+    periods = [get_period(date, days=60) for date in dates]
 
     try:
-        filtered_collections_s2c = [
-            filter_collection(collection_s2c, coords, p) for p in periods
-        ]
-        patches_s2c = [
-            get_patch_s2(c, coords, radius, bands=bands_s2c, crop=crop_s2c)
-            for c in filtered_collections_s2c
+        filtered_collections = [
+            filter_collection(collection, coords, p) for p in periods
         ]
-        filtered_collections_s2a = [
-            filter_collection(collection_s2a, coords, p) for p in periods
+        patches = [
+            get_patch(c, coords, radius, bands=bands, crop=crops, dtype=dtype)
+            for c in filtered_collections
         ]
-        patches_s2a = [
-            get_patch_s2(c, coords, radius, bands=bands_s2a, crop=crop_s2a)
-            for c in filtered_collections_s2a
-        ]
-        filtered_collections_s1 = [
-            filter_collection_s1(collection_s1, coords, p) for p in periods
-        ]
-        patches_s1 = [
-            get_patch_s1(c, coords, radius, bands=bands_s1, crop=crop_s1)
-            for c in filtered_collections_s1
-        ]
-
-        center_coord = coords
 
     except (ee.EEException, urllib3.exceptions.HTTPError) as e:
         if debug:
             print(e)
-        return None, None, None, coords
+        return None, coords
 
-    return patches_s1, patches_s2c, patches_s2a, center_coord
+    return patches, coords
 
 
 def save_geotiff(
```

Comment:
I know @calebrob6 had a question about image compression. We should quickly benchmark to see whether no compression or high compression is better for I/O. Note that HPC clusters tend to have extremely slow filesystems, so a smaller compressed file may actually be faster to load even if it needs to be decompressed. Also makes downloads faster. I know you guys used HDF5 or some other file format for faster loading, but I think we'll want to keep geospatial metadata if possible and make the files easy to load.

Reply:
I had to get our inode limit increased, but we might be okay. I'm hoping to avoid this but we'll see if it's too slow.